index.js

```diff
@@ -37,33 +63,58 @@ const getLuminance = (RGBarray) => {
   return luminance;
 };
 
-const twoHexesRatio = (color1, color2) => {
-  const RGBColor1 = hexToRGB(color1);
-  const RGBColor2 = hexToRGB(color2);
-  const color1Luminance = getLuminance(RGBColor1);
-  const color2Luminance = getLuminance(RGBColor2);
+const calculateRatio = (color1, color2) => {
+  const color1Luminance = getLuminance(color1);
+  const color2Luminance = getLuminance(color2);
   /* (L1 + 0.05) / (L2 + 0.05), whereby:
     L1 is the relative luminance of the lighter of the colors, and
     L2 is the relative luminance of the darker of the colors. */
-  let [lighterLum, darkerLum] = [color1Luminance, color2Luminance];
-  if (darkerLum > lighterLum) {
-    [lighterLum, darkerLum] = [darkerLum, lighterLum];
-  }
+  let lighterLum = Math.max(color1Luminance, color2Luminance);
+  let darkerLum = Math.min(color1Luminance, color2Luminance);
   return ((lighterLum + 0.05) / (darkerLum + 0.05)).toFixed(2);
 };
 
+const twoHexesRatio = (color1, color2) => {
+  const RGBColor1 = hexToRGB(color1);
+  const RGBColor2 = hexToRGB(color2);
+  return calculateRatio(RGBColor1, RGBColor2);
+};
+
+const twoRgbsRatio = (color1, color2) => {
+  const RGBColor1 = rgbToRGB(color1);
+  const RGBColor2 = rgbToRGB(color2);
+  return calculateRatio(RGBColor1, RGBColor2);
+};
+
+const twoRgbasRatio = (color1, color2) => {
+  const RGBColor1 = rgbaToCloseRGB(color1);
+  const RGBColor2 = rgbaToCloseRGB(color2);
+  return calculateRatio(RGBColor1, RGBColor2);
+};
+
 const displayResult = () => {
   let firstColor = foregroundColor.value;
   let secondColor = backgroundColor.value;
+  const rgbRegex = /^rgb.*/i;
+  const rgbaRegex = /^rgba.*/i;
   if (firstColor.length === 7) {
     foregroundSwatch.style.backgroundColor = firstColor;
   }
   if (secondColor.length === 7) {
     backgroundSwatch.style.backgroundColor = secondColor;
   }
+  // CASE two Hexes
   if (firstColor.length === 7 && secondColor.length === 7) {
     ratioResult.innerHTML = twoHexesRatio(firstColor, secondColor);
   }
+  // CASE two RGBAs
+  else if (rgbaRegex.test(firstColor) && rgbaRegex.test(secondColor)) {
+    ratioResult.innerHTML = twoRgbasRatio(firstColor, secondColor);
+  }
+  // CASE two RGBs
+  else if (rgbRegex.test(firstColor) && rgbRegex.test(secondColor)) {
+    ratioResult.innerHTML = twoRgbsRatio(firstColor, secondColor);
+  }
```

Comment:
We should show the color swatches for the rgb and rgba conditions since we already have that implemented for the hex condition.

For example
```
 // CASE two RGBs
  if (rgbRegex.test(firstColor) && rgbRegex.test(secondColor)) {
    foregroundSwatch.style.backgroundColor = firstColor;
    backgroundSwatch.style.backgroundColor = secondColor;
    ratioResult.innerHTML = twoRgbsRatio(firstColor, secondColor);
  }
```
will produce this result
<img width="684" alt="Screen Shot 2022-08-14 at 10 10 08 PM" src="https://user-images.githubusercontent.com/67210629/184579919-eea217ce-bb27-49c5-b9b8-125705965d17.png">